File: 0977: squaresOfASortedArray.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [0]*len(nums)
        low, high = 0, len(nums)-1
        while low <= high:
            left, right = abs(nums[low]), abs(nums[high])
            if left > right:
                res[high-low] = left*left
                low += 1
            else:
                res[high-low] = right*right
                high -= 1
        return res

File: 0977: squaresOfASortedArray.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        priority_queue<int> ans;
        for(int i = 0; i < nums.size(); i++){

            ans.push(nums[i]*nums[i]);
        }

        int i = nums.size()-1;
        while(!ans.empty()){
            nums[i] = ans.top();
            ans.pop();
            i--;
        }

        return nums;
    }
};